Prepare for transformers v5 upgrade

winglian opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Expanded build pipeline support for new dependency versions: CUDA 12.8.1, Python 3.11, PyTorch 2.9.0, and PyTorch 2.9.1. Updated both standard and cloud build configurations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

winglian committed:
Prepare for transformers v5 upgrade

winglian committed:
fix hf cli

winglian committed:
update for hf hub changes

salmanmohammadi commented on .github/workflows/tests.yml:
```diff
@@ -109,7 +109,7 @@ jobs:
       - name: Pre-Download dataset fixture
         run: |
-          huggingface-cli download --repo-type=dataset axolotl-ai-internal/axolotl-oss-dataset-fixtures
+          hf download --repo-type=dataset axolotl-ai-internal/axolotl-oss-dataset-fixtures
```
Have they removed huggingface-cli in v5?

winglian replied:
I think they bumped the minimum version of hf hub in transformers that forces this change

winglian committed:
fix tokenizer apply_chat_template args

winglian committed:
remap include_tokens_per_second

NanoCode012 commented on tests/hf_offline_utils.py:
```diff
@@ -6,22 +6,21 @@
     importlib.reload(huggingface_hub.constants)
     huggingface_hub.constants.HF_HUB_OFFLINE = hf_hub_offline
     importlib.reload(datasets.config)
     datasets.config.HF_HUB_OFFLINE = hf_hub_offline
```
Seems like we're hitting HF OFFLINE errors in our tests

winglian replied:
yeah, they swapped out the underlying implementation to requests, but we had previously been able to workaround it

winglian committed:
fix tps

winglian committed:
handle migration for warmup

winglian committed:
use latest hf hub

winglian committed:
Fix scan -> ls

winglian committed:
fix import

winglian committed:
fix for renaming of mistral common tokenizer -> backend

winglian committed:
update for fixed tokenziation for llama

winglian committed:
Skip phi35 tests for now

winglian committed:
remove mistral patch fixed upstream in huggingface/transformers#41439

winglian committed:
use namespacing for patch

winglian committed:
don't rely on sdist for e2e tests for now

winglian committed:
run modal ci without waiting too

winglian committed:
Fix dep for ci

winglian committed:
fix imports

winglian committed:
Fix fp8 check

winglian committed:
fsdp2 fixes

winglian committed:
fix version handling

winglian committed:
update fsdp version tests for new v5 behavior

winglian committed:
Fail multigpu tests after 3 failures

winglian committed:
skip known v5 broken tests for now and cleanup

salmanmohammadi commented on requirements.txt:
```diff
@@ -10,14 +10,14 @@ liger-kernel==0.6.4
 tokenizers>=0.22.1
-transformers==4.57.1
+transformers @ git+https://github.com/huggingface/transformers.git@main
 accelerate==1.11.0
```
I think we need to bump the accelerate version @winglian

winglian committed:
bump deps

winglian committed:
unmark skipped test

winglian committed:
re-enable test_fsdp_qlora_prequant_packed test

winglian committed:
increase multigpu ci timeout

winglian committed:
skip broken gemma3 test

winglian committed:
reduce timout back to original 120min now that the hanging test is skipped

winglian committed:
fix for un-necessary collator for pretraining with bsz=1

NanoCode012 commented:
Require https://github.com/axolotl-ai-cloud/axolotl/pull/3318

NanoCode012 committed:
fix: safe_serialization deprecated in transformers v5 rc01 (#3318)

winglian committed:
torch_dtype deprecated

winglian committed:
load model in float32 for consistency with tests

winglian committed:
revert some test fixtures back

winglian committed:
use hf cache ls instead of scan

winglian committed:
don't strip fsdp_version

more fdsp_Version fixes for v5
fix version in fsdp_config
fix aliasing
fix fsdp_version check
check fsdp_version is 2 in both places

salmanmohammadi committed:
Transformers v5 rc2 (#3347)

* bump dep

* use latest fbgemm, grab model config as part of fixture, un-skip test

* import AutoConfig

* don't need more problematic autoconfig when specifying config.json manually

* add fixtures for argilla ultrafeedback datasets

* download phi4-reasoning

* fix arg

* update tests for phi fast tokenizer changes

* use explicit model types for gemma3

---------

Co-authored-by: Wing Lian <wing@axolotl.ai>

NanoCode012 committed:
fix: AutoModelForVision2Seq -> AutoModelForImageTextToText

NanoCode012 committed:
chore: remove duplicate

NanoCode012 committed:
fix: attempt fix gemma3 text mode

NanoCode012 committed:
chore: lint

winglian committed:
ga release of v5

winglian committed:
need property setter for name_or_path for mistral tokenizer

winglian committed:
vllm not compatible with transformers v5

winglian committed:
setter for chat_template w mistral too
